File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2022w/ID07IO1/Command/src/SportsmenCommandTrening.java

import java.util.List;

public class SportsmenCommandTrening implements SportsmenCommand
{
   private final List<SportsmenCommand> cmds;
   public SportsmenCommandTrening(List<SportsmenCommand> cmds)
   {
      this.cmds=cmds;
   }
   @Override public String execute(Sportsmen sportsmen)
   {
      StringBuilder sb=new StringBuilder();
      boolean first=true;
      for(SportsmenCommand cmd:cmds)
      {
         if(first) first=false; else sb.append(", ");
         sb.append(cmd.execute(sportsmen));
      }
      return sb.toString();
   }
}

File: semestr-8/Jezyk C#/alexander/Csharp/Archiwum/2022s/ID06IO1/Command/Command/SportsmenCommandTrening.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Command
{
    internal class SportsmenCommandTrening: SportsmenCommandComplex
    {
        private readonly List<SportsmenCommand> cmds;
        public SportsmenCommandTrening(string name, List<SportsmenCommand> cmds) :base(name)
        {
            this.cmds = cmds;
        }
        public override string execute(Sportsmen sportsmen)
        {
            StringBuilder sb = new StringBuilder();
            /*
            bool first = true;
            foreach(SportsmenCommand cmd in cmds)
            {
                //if (!first) sb.Append(", ");
                //first = false;
                if (first) first = false; else sb.Append(", ");
                sb.Append(cmd.execute(sportsmen));
            }
            */
            for(int i=0;i<cmds.Count;++i)
            {
                if (i>0) sb.Append(", ");
                sb.Append(cmds[i].execute(sportsmen));
            }
            return "{ "+sb.ToString()+" }";
        }
    }
}
